caffe2/CMakeLists.txt

```diff
@@ -141,6 +141,17 @@ if(BUILD_CAFFE2 AND (NOT INTERN_BUILD_MOBILE OR BUILD_CAFFE2_MOBILE))
   add_subdirectory(mpi)
   add_subdirectory(observers)
   add_subdirectory(onnx)
+
+  include_directories(../third_party/optimizer)
+
+  set(ONNX_SRCS
+    "../third_party/optimizer/onnxoptimizer/optimize.cc"
+    "../third_party/optimizer/onnxoptimizer/pass.cc"
+    "../third_party/optimizer/onnxoptimizer/pass_registry.cc"
+    "../third_party/optimizer/onnxoptimizer/pass_manager.cc")
+
+  list(APPEND Caffe2_CPU_SRCS ${ONNX_SRCS})
```

Comment:
Please move this code to `cmake/Dependencies.cmake` next to the other ONNX dependencies logic.
And is there a reason why we need to explicitly specify source files rather than just make caffe2 depend on  `onnx_optimizer` library?

Reply:
@malfet 
Thanks, I'm trying to include the Optimizer library in the Dependencies.cmake. But I'm not sure how to make this work where USE_SYSTEM_ONNX is set. How can we make sure Optimizer is set up in such environments?

Reply:
@malfet 
The onnx optimizer library also depends on ONNX and this is causing a dependency cycle. What do you think?

Reply:
Let's not worry about `USE_SYSTEM_ONNX` for the time being. But I'm not sure what do you mean about cyclic dependency? This looks like a DAG to me, isn't i, i.e. libtorch_cpu depends on onnx and onnxoptimizer, while onnxoptimizer depends on onnx,  i.e.:
```
 libtorchcpu <--- ONNX
     ^             ^
     |             |
     |---- ONNXoptimizer
```

Reply:
So ONNX is a submodule of the optimizer. In this case, wouldn't it look like:
```
 libtorchcpu +--- ONNX
     +           |
     |           +
     |---- ONNXoptimizer